### PR TITLE
Fixes Issue #428,  #479, #546, #551 - add support for manifest yaml file, make preprocessors simple

### PR DESCRIPTION
**Pull request by richmahn**
This is based off the python-resource-container (https://github.com/unfoldingWord-dev/python-resource-container) keeping the "RC" class name, but greatly enhanced to take data from difference sources (package.json and manifest.json of older pre-rc version, or the repo name itself if those don't exist) and aid in the preprocessing done by the client.

#479, #546, #551 -  just because preprocessing had to be done very different, dealing with multiple projects in a RC, and different project directories

### CODE REVIEW

File: client/client_webhook.py [hunk 1]
```diff
@@ -83,69 +82,29 @@ def process_webhook(self):
...
         zip_filepath = os.path.join(self.base_temp_dir, zip_filename)
         self.logger.info('Zipping files from {0} to {1}...'.format(output_dir, zip_filepath))
         add_contents_to_zip(zip_filepath, output_dir)
-        if os.path.isfile(manifest_path) and not os.path.isfile(os.path.join(output_dir, 'manifest.json')):
-            add_file_to_zip(zip_filepath, manifest_path, 'manifest.json')
         self.logger.info('finished.')
 
         # 4) Upload zipped file to the S3 bucket
-        s3_handler = self.s3_handler_class(self.pre_convert_bucket)
+        s3_handler = S3Handler(self.pre_convert_bucket)
```
Comment (PhotoNomad0) [inline client/client_webhook.py:107]: Why is this hard coded to S3Handler, wasn't self.s3_handler_class there for testing?

Comment (richmahn) [reply client/client_webhook.py:107]: Nope, now that we know how to mock stuff (I didn't when I wrote the original code) we don't need to make our own mocking classes.

Comment (richmahn) [reply client/client_webhook.py:107]: You can mock a function of an existing class, change its return value, etc, as you probably know.

Comment (richmahn) [reply client/client_webhook.py:107]: Or more importantly, we use moto, for mocking boto, which we can set up our own S3 directory structure which actually serves boto3 S3 calls. https://github.com/unfoldingWord-dev/tx-manager/blob/develop/tests/door43_tools_tests/test_project_deployer.py#L46

**Review by PhotoNomad0 — APPROVED** [review]

File: resource_container/ResourceContainer.py [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+        else:
+            return datetime.utcnow().strftime("%Y-%m-%d")
+
+    @property
+    def modified(self):
+        return self.resource.get('modified', datetime.utcnow().strftime("%Y-%m-%d"))
+
+    @property
+    def rights(self):
+        return self.resource.get('rights', 'CC BY-SA')
```
Comment (da1nerd) [inline resource_container/ResourceContainer.py:391]: I think we have always been specifying the version: `CC BY-SA 4.0`

Comment (richmahn) [reply resource_container/ResourceContainer.py:391]: Ok, will update. Not sure which manifest example I got that from. Maybe the readthedocs?

Comment (richmahn) [reply resource_container/ResourceContainer.py:391]: Hm, nope, it is CC BY-SA 4.0 there too

**Pull request merged**